src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GoalController.kt

```diff
@@ -165,13 +165,28 @@ class GoalController(
   fun getGoals(
     @PathVariable @Pattern(regexp = PRISON_NUMBER_FORMAT) prisonNumber: String,
     @RequestParam(required = false, name = "status") statuses: Set<GoalStatus>?,
-  ) =
-    GetGoalsResponse(
+  ): GetGoalsResponse {
+    val response = GetGoalsResponse(
       goalService.getGoals(
         GetGoalsDto(prisonNumber, convertStatuses(statuses)),
       ).map { goal -> goalResourceMapper.fromDomainToModel(goal) },
     )
 
+    // Map each goal response to its corresponding version with notes
```

Comment:
This is the core fix, and is basically what we do in the ActionPlan controller when getting the Action Plan (which includes Goals)